src/Formatter/UnparsePostMentions.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Mentions\Formatter;
+
+use s9e\TextFormatter\Utils;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class UnparsePostMentions
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * Configure rendering for user mentions.
+     *
+     * @param string $xml
+     * @param mixed $context
+     * @return string $xml to be unparsed
+     */
+    public function __invoke($context, string $xml)
+    {
+        $xml = $this->updatePostMentionTags($context, $xml);
+        $xml = $this->removePostMentionTags($xml);
+
+        return $xml;
+    }
+
+    protected function updatePostMentionTags($context, string $xml)
+    {
+        $post = $context;
+
+        return Utils::replaceAttributes($xml, 'POSTMENTION', function ($attributes) use ($post) {
+            $post = $post->mentionsPosts->find($attributes['id']);
+            if ($post && $post->user) {
+                $attributes['displayname'] = $post->user->display_name;
+            }
+
+            if (! $post) {
+                $attributes['displayname'] = $this->translator->trans('flarum-mentions.forum.post_mention.unknown_text');
+            }
+
+            if ($post && ! $post->user) {
+                $attributes['displayname'] = $this->translator->trans('core.lib.username.deleted_text');
+            }
+
+            if (strpos($attributes['displayname'], '"#') !== false) {
+                $attributes['displayname'] = preg_replace('/"#[a-z]{0,3}[0-9]+/', '_', $attributes['displayname']);
+            }
+
+            return $attributes;
+        });
+    }
+
+    protected function removePostMentionTags(string $xml)
```

Comment:
What's this method (and the corresponding one in `UnparseUserMentions`) for?

Reply:
So basically, the normal unparsing process just strips the `$xml` of all xml tags, leaving only their content, i.e
`<USERMENTION displayname="PotatoDev" id="1">@SychO</USERMENTION>` => `@SychO`

So in these invokable classes, we strip the mention tags ourselves, by running a preg_replace that essentially does this:

`<USERMENTION displayname="PotatoDev" id="1">@SychO</USERMENTION>` => `@"PotatoDev"#1`

 to update the raw content's displayname and format at the same time, that's why we run an update on the tags first and then remove, so that the displayname inside is updated.

Reply:
we do though

Reply:
Ah, I just can't read. Excellent then, carry on! 

Maybe the methods could be renamed to be a bit clearer as to the goal?

Reply:
Yeah true, it's unclear what it really does, will do.